x-pack/solutions/security/plugins/security_solution/public/data_view_manager/containers/data_view_provider.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { FC, PropsWithChildren } from 'react';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { useKibana } from '../../common/lib/kibana';
+import { createDefaultDataView } from '../utils/create_default_data_view';
+
+const DefaultDataViewContext = createContext<DataView | undefined>(undefined);
+
+export const useDefaultDataView = () => {
+  const defaultDataView = useContext(DefaultDataViewContext);
+
+  if (!defaultDataView) {
+    throw new Error('Could not fetch default data view');
+  }
+
+  return defaultDataView;
+};
+
+export const DefaultDataViewProvider: FC<PropsWithChildren> = ({ children }) => {
+  const { dataViews, uiSettings, spaces, application, http } = useKibana().services;
+
+  const [defaultDataView, setDefaultDataView] = useState<DataView>();
+
+  useEffect(() => {
+    (async () => {
```

Comment:
Even though this _should_ only run once, we should wrap it in an `if (!defaultDataView) { (async () => {})() }`